src/ifcgeom/IfcGeomIterator.h

```diff
@@ -479,23 +591,45 @@ namespace IfcGeom {
 		}
 
 		bool create() {
+			bool success = true;
+
+			IfcGeom::BRepElement<P>* next_shape_model = 0;
+			IfcGeom::SerializedElement<P>* next_serialization = 0;
+			IfcGeom::TriangulationElement<P>* next_triangulation = 0;
+
 			try {
-				current_shape_model = create_shape_model_for_next_entity();
+				next_shape_model = create_shape_model_for_next_entity();
 			} catch (...) {}
-			if (!current_shape_model) return false;
-            if (settings.get(IteratorSettings::USE_BREP_DATA)) {
-				try {
-					current_serialization = new SerializedElement<P>(*current_shape_model);
-				} catch (...) {}
-				return !!current_serialization;
-            } else if (!settings.get(IteratorSettings::DISABLE_TRIANGULATION)) {
-				try {
-					current_triangulation = new TriangulationElement<P>(*current_shape_model);
-				} catch (...) {}
-				return !!current_triangulation;
+
+			if (next_shape_model) {
+				if (settings.get(IteratorSettings::USE_BREP_DATA)) {
+					try {
+						next_serialization = new SerializedElement<P>(*next_shape_model);
+					} catch (...) {
+						success = false;
+					}
+				} else if (!settings.get(IteratorSettings::DISABLE_TRIANGULATION)) {
+					try {
+						if (ifcproduct_iterator == ifcproducts->begin() || settings.get(IteratorSettings::USE_WORLD_COORDS)) {
+							next_triangulation = new TriangulationElement<P>(*next_shape_model);
+						} else {
+							next_triangulation = new TriangulationElement<P>(*next_shape_model, current_triangulation->geometry_pointer());
```

Comment:
@Stinkfist0 This does not work after merging, because with your `take ownership` flag `current_triangulation == 0`. Is this flag still necessary as the min,max bounds are now computed during initialization?